pubsub/query/parser_test.go

```diff
@@ -0,0 +1,53 @@
+package query_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tmlibs/pubsub/query"
+)
+
+// TODO: fuzzy testing?
+func TestParser(t *testing.T) {
+	cases := []struct {
+		query string
+		valid bool
+	}{
+		{"tm.events.type=NewBlock", true},
+		{"tm.events.type = NewBlock", true},
+		{"tm.events.type>NewBlock", false},
+		{"", false},
+		{"=", false},
+		{"=NewBlock", false},
+		{"tm.events.type=", false},
+
+		{"tm.events.typeNewBlock", false},
+		{"NewBlock", false},
+		{"", false},
+
+		{"tm.events.type=NewBlock AND", false},
+		{"AND tm.events.type=NewBlock ", false},
+		{"tm.events.type=NewBlock AND abci.account.name=Igor", true},
+
+		{"abci.account.name CONTAINS Igor", true},
+
+		{"tx.date > DATE 2013-05-03", true},
+		{"tx.date < DATE 2013-05-03", true},
+		{"tx.date <= DATE 2013-05-03", true},
+		{"tx.date >= DATE 2013-05-03", true},
+
+		{"tx.date > TIME 2013-05-03T14:45:00+07:00", true},
+		{"tx.date < TIME 2013-05-03T14:45:00-02:00", true},
+		{"tx.date <= TIME 2013-05-03T14:45:00Z", true},
+		{"tx.date >= TIME 2013-05-03T14:45:00Z", true},
```

Comment:
can we add some `false` cases for DATE/TIME ? also would be good to add cases for ints/floats (including false cases like `43.21.1` ... or is that just a string ? ). Also false cases for things like `AN` and `TIM`, and `==` on strings ... edge-cases.